infrastructure/src/main/java/org/corfudb/infrastructure/logreplication/transport/sample/GRPCLogReplicationServerHandler.java

```diff
@@ -118,7 +118,7 @@ public void send(ResponseMsg msg) {
                 // the one for which a response is being sent.
                 replicationStreamObserverMap.keySet().removeIf(id -> id <= requestId);
             } catch (Exception e) {
-                log.error("Caught exception while trying to send message {}", msg.getHeader().getRequestId());
+                log.error("Caught exception while trying to send message {}", msg.getHeader().getRequestId(), e);
```

Comment:
What if there is a connection error, would we spam logs with exceptions?

Reply:
when a connection error will appear, your code starts printing exceptions in the log, and I guess there will be a lot of them? I'm worried if the log would be polluted with exceptions

Reply:
It's very expensive to print exceptions in the log, they are huge multiline messages

Reply:
For this file it's only a sample implementation used in tests. I thought it might be worse if we just don't know what the exception is from the log?

btw, I noticed lots of similar usage pattern in our code base, i.e. `log.error("", e)`. I thought compared to the additional lines of logs, it might be rational to print the body upon exception so we at least know them?

Reply:
I see, ok